txselector/txselector.go

```diff
@@ -472,10 +472,31 @@ func (txsel *TxSelector) processL2Txs(
 			break
 		}
 
+		// Check if MaxNumBatch is exceeded
+		if l2Txs[i].MaxNumBatch > uint32(txsel.localAccountsDB.CurrentBatch())+1 {
```

Comment:
Maybe here it would be:
```if l2Txs[i].MaxNumBatch < uint32(txsel.localAccountsDB.CurrentBatch())+1 { ```
because if the maximum batch number user wanted is 100 and current batch is 101 so it is become invalid. User wants his transaction processed before the current batch so if MaxNumBatch is 100 and the `uint32(txsel.localAccountsDB.CurrentBatch())+1` is 101 so the transaction is invalid.

Reply:
It may be a bit confusing because of the comment, but if the expression is evaluated to true the tx is rejected.
Also `txsel.localAccountsDB.CurrentBatch()` return the last processed batch, but here we're running the selection for the next batch, that's the reason for the `+1`